### DISCUSSION TITLE
Rendering pdf with Charts (Charts.js)

### DISCUSSION THREAD

**Discussion by tal-VSL** [diegomura/react-pdf#1784] [Q&A]
Hi,
I'm trying to publish a pdf file with various charts (pie & line charts).
for example https://react-chartjs-2.js.org/examples/pie-chart
There is a way to import my charts components and render them into a pdf file?

Waiting for an update,
Thanks!

**Comment by crobinson42** [comment]
@tal-VSL as you can see in the screenshot the chart is drawn in a <canvas> element. react-pdf uses yoga layout and react-native-style components to map to the pdf elements under the hood.

What I would suggest to anyone going down this path is 2 options (disclaimer: I haven't done either, yet):

1. Use react-pdf <Svg/> component and generate your own charts
2. If you want something like react-chartjs, render the chart, get the <canvas> element and create a base64 datauri _snapshot_ then use the react-pdf <Image/> component and pass the base64 datauri image as the src.

<img width="1248" alt="image" src="https://github.com/user-attachments/assets/d48aea5b-5bd4-4a28-9523-183cd751b122" />
